qucs-core/adms/admsXml/admsCheck.1
.TH admsCheck "1" "September 2008" "Debian/GNU Linux" "User Commands"
.SH NAME
admsCheck \- Test Program for admsXml
.SH SYNOPSIS
.B admsCheck
.SH DESCRIPTION
admsCheck creates two files: dummy.va dummy.xml.
These two files illustrate how program admsXml works.
dummy.va is an example of Verilog-AMS file.
dummy.xml is a control file used by admsXml to convert file dummy.va into file dummy.net.
More specifically running 'admsXml -e dummy.xml dummy.va' creates file dummy.net.
The contents of file dummy.net is entirely controlled by the instructions provided in file dummy.xml. 
.SH FILES
dummy.va: dummy Verilog-AMS file created by admsCheck.
dummy.xml: dummy xml file created by admsCheck.
dummy.net: file created by admsXml.
.SH OPTIONS
There is no options for this program.
.SH AVAILABILITY
The latest version of admsCheck can always be obtained from \fBmot-adms.sourceforge.net\fR
.SH "REPORTING BUGS"
Known bugs are documented within the BUGS file.  Report bugs to
<r29173@users.sourceforge.net>.
.SH COPYRIGHT
.PP
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
.SH AUTHORS
Written by Laurent Lemaitre <r29173@users.sourceforge.net>.


NEWS
--
-- NEWS
--
-- Copyright (C) 2003-2011 Stefan Jahn <stefan@lkcc.org>
--
-- This is free software; you can redistribute it and/or modify
-- it under the terms of the GNU General Public License as published by
-- the Free Software Foundation; either version 2, or (at your option)
-- any later version.
--
-- This software is distributed in the hope that it will be useful,
-- but WITHOUT ANY WARRANTY; without even the implied warranty of
-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-- GNU General Public License for more details.
--
-- You should have received a copy of the GNU General Public License
-- along with this package; see the file COPYING.  If not, write to
-- the Free Software Foundation, Inc., 51 Franklin Street - Fifth Floor,
-- Boston, MA 02110-1301, USA.
--

This file lists user visible changes that were made between releases.  For
more verbose descriptions of these and internal changes see the ChangeLog
files.

Version 0.0.16
--------------

 * implementation of interactive GNU Octave connection
 * support for C++ code export of symbol drawings associated with
   Verilog-A files
 * direct association of symbol drawings to Verilog-HDL, Verilog-A
   and VHDL code
 * added Verilog-HDL and Verilog-A syntax highlighting in text documents
 * support for pre-compiled VHDL modules and libraries made from user
   written VHDL code
 * new components: NIGBT, HICUM L2 v2.24, HICUM L0 v1.2g, tunnel diode,
   ideal coupled transmission line, hybrid, HICUM L0 v1.3
 * translation into Kazakh
 * allow equations in Verilog-HDL and VHDL subcircuits

Version 0.0.15
--------------

 * passing parameters to Verilog-HDL subcircuits allowed
 * VHDL files and subcircuits can handle typed generic parameters
 * translation into Arabic
 * import dialog replaced by complete import/export frontend for the
   Qucs-Converter command line tool
 * allowing VHDL files on schematic with arbitrary in/out signals
 * many new digital primitives defined for analogue, VHDL and Verilog
   simulations
 * additional install step for MacOSX and appropriate icons
 * US letter formats for schematic frame
 * simulation time for digital files (Verilog and VHDL) now stored in
   additional configuration file
 * Czech translation of the internal help system
 * new components: EPFL-EKV NMOS/PMOS V2.6, rectangular waveguide,
   HICUM L0 v1.2, HICUM L2 v3.23, photodiode, digital buffer, microstrip
   radial stub, phototransistor, RLCG transmission line
 * new libraries added: PMOSFETs, NMOSFETs, Regulators, Varistors and
   Ideal

Version 0.0.14
--------------

 * added coplanar line types to Qucs-Transcalc
 * fixed printing (font sizes) under Win32
 * support for sub- and super-script in graphical text paintings
 * new components: diac, triac, thyristor, logarithmic amplifier,
   HICUM L0 v1.12, potentiometer, equation defined RF device, MESFET

Version 0.0.13
--------------

 * new components: file based current and voltage sources, modular
   opamp, HICUM L2 v2.22

Version 0.0.12
--------------

 * preprocessor selectable in SPICE file component
 * translation into Ukrainian
 * libraries can now contain analog as well as digital subcircuits
 * support for symbolically defined devices
 * new components: exponential voltage and current source
 * added Verilog file component
 * support for digital simulations using Verilog

Version 0.0.11
--------------

 * new devices: HICUM L2 v2.1 and FBH HBT
 * allow equations in subcircuits
 * subcircuits parameters
 * translation into Czech and Catalan
 * tabular scrollbar can be moved directly with mouse cursor dragging
 * better painting and doing with zoom factors > 1
 * some bug fixes
 * some internal restructuring
 * new components in GUI and in library

Version 0.0.10
--------------

 * optimization dialog for ASCO added, preliminary support
 * added an attenuator synthesis application
 * many, many small bug fixes and improvements
 * support for nine-valued VHDL logic
 * support for user libraries created from subcircuits
 * import of Touchstone, CITI, VCD and MDL data files
 * export data to CSV files
 * creation of project package file for data exchange
 * schematic frame as document property
 * syntax highlighting for VHDL data types and attributes

Version 0.0.9
-------------

 * components can be either deactivated as short or open
 * VHDL files can be used as subcircuits
 * VHDL text editor
 * translation into Russian
 * number of ports of S-parameter component not anymore limited
 * language can be chosen in application settings dialog
 * new components: coaxial line, differential voltage probe, switch, AM-
   and PM-modulator, relais
 * fixed bug in Cauer band-stop filter

Version 0.0.8
-------------

 * digital simulation using FreeHDL is finally working
 * translation into Turkish
 * new diagrams: truth table and timing diagram
 * add non-Qucs files to project
 * dialog for changing properties of several components
 * creating matching circuits
 * more filters supported by filter synthesis tool
 * many more models in component library
 * new components: digital gates, correlated noise sources,
   mutual inductors
 * printing DC bias into schematic
 * hide invisible lines in 3D diagram

Version 0.0.7
-------------

 * text painting rotations in subcircuit symbols
 * renamed Qucs, new name is "Quite Universal Circuit Simulator"
 * several small improvements and bug fixes
 * Latex-like mnemonics (e.g. \tau) in graphic texts and diagram labels
 * library manager including transistor, diode, substrate, etc. libraries

Version 0.0.6
-------------

 * finished SPICE component
 * transmission line calculator
 * filter synthesis application
 * new diagrams: 3-dimensional cartesian, locus curve
 * many small improvements
 * edit component properties directly on the schematic
 * translations into Swedish
 * new components: coplanar open, short, gap and step
 * nicer icons for simulation components

Version 0.0.5
-------------


qucs-core/adms/admsXml/admsCheck.c
/*
 * RCS Info
 * $Id: admsCheck.c 1110 2008-10-14 13:40:29Z r29173 $
 * 
 * Log
 * $Log$
 * Revision 1.17  2006/09/13 16:13:20  r29173
 * started migration to www.antlr.org tools
 *
 * Revision 1.16  2006/08/04 17:15:32  r29173
 * rearranged code partionning into files
 *
 * Revision 1.15  2006/06/16 12:24:25  r29173
 * ton of changes: memory clean-up
 *
 * Revision 1.14  2006/06/09 16:49:34  r29173
 * improved malloc management
 *
 * Revision 1.13  2006/01/26 13:39:25  r29173
 * fixed core dump when running admsCheck
 *
 * Revision 1.12  2005/06/21 08:02:28  r29173
 * renamed adms_set_admsmain to adms_valueto_admsmain
 *
 * Revision 1.11  2005/05/09 14:38:29  r29173
 * cleaned-up source directory structure
 *
 * Revision 1.10  2005/05/03 09:35:14  r29173
 * cleaned-up header file dependencies
 *
 * Revision 1.9  2005/05/03 07:58:19  r29173
 * moved admsNS into admsFile
 *
 * Revision 1.8  2005/04/01 07:25:44  r29173
 * renamed prefix .xsl to .xml
 *
 * Revision 1.7  2005/01/11 09:58:01  r29173
 * tons of changes - simplify accessor names (valueof, valueto)
 *
 * Revision 1.6  2004/11/19 12:29:56  r29173
 * renamed _C into CAT
 * saved argc, argv into admsmain element
 * added adms:setenv transform
 *
 * Revision 1.5  2004/11/11 13:45:03  r29173
 * replaced g_message by adms_message_info
 *
 * Revision 1.4  2004/11/01 09:37:36  r29173
 * changed messaging macros (avoid useless computation of arguments)
 * improved the coding of transforms in xmlProgram.c
 *
 * Revision 1.3  2004/08/03 12:33:54  r29173
 * import adms-1.21.0 from local CVS
 *
 * Revision 1.2  2004/08/03 10:05:46  r29173
 * removed superelement from admsgui
 * fixed some warnings issued during compilation
 *
 * Revision 1.1.1.1  2004/05/21 12:20:01  r29173
 * recreated cvs data structure (crashed after revision 1.13.0!)
 *
 * Revision 1.10  2004/03/11 16:25:14  r29173
 * started implementing code builder in yacc parser
 *
 * Revision 1.9  2004/03/08 13:58:05  r29173
 * all code lower-cased
 *
 * Revision 1.8  2004/03/08 08:24:19  r29173
 * started implementing code builder in yacc parser
 *
 * Revision 1.7  2004/01/13 11:25:51  r29173
 * added adms_message_usage
 *
 * Revision 1.6  2004/01/09 16:23:45  r29173
 * cleaned-up use of #include
 *
 * Revision 1.5  2004/01/08 10:28:49  r29173
 * moved code to always.[ch]
 *
 * Revision 1.4  2004/01/05 11:07:51  r29173
 * change the way admsCheck.xml is downloaded
 *
 * Revision 1.3  2003/10/22 15:11:36  r29173
 * changed the way script "text2Ccode.pl" handles file name.
 *
 * Revision 1.2  2003/05/21 14:18:00  r29173
 * add rcs info
 *
 */

#include "adms.h"
/*constants*/
#include "dummy.xml.c"
#include "dummy.va.c"

char*mydata[]={
"dummy.va","dummy_va",
"dummy.xml","dummy_xml",
NULL
};

int main(const int argc, const char **argv)
{
  char**mydataptr;
  rootnew(adms_admsmain_new("admsmain"));
  adms_admsmain_valueto_filename(root(),adms_kclone("stdin"));
  adms_message_info(("Create files:\n"));
  for( mydataptr=mydata; *mydataptr; mydataptr++,mydataptr++)
  {
    char*savefilename=*mydataptr;
    char*symbolname=*(mydataptr+1);
    char* myfilename=strdup(savefilename);
    FILE* ofh=fopen(myfilename,"w");
    adms_message_info((" ... %s\n",myfilename));
    free(myfilename);
    if(!strcmp(symbolname,"dummy_va"))
      fprintf(ofh,"%s",dummy_va);
    else if(!strcmp(symbolname,"dummy_xml"))
      fprintf(ofh,"%s",dummy_xml);
    fclose(ofh);
  }
  adms_message_info(("How to use the example\n"));
  adms_message_info(("  admsXml dummy.va -e dummy.xml\n"));
  adms_message_info(("  This command parses file dummy.va\n"));
  adms_message_info(("  and interprets control file dummy.xml\n"));
  adms_message_info(("  File dummy.net is created.\n"));
  adms_message_info(("  (file .dummy.va.adms is created in background.)\n"));
  return 0;
}
